src/core/strings.rs

```diff
@@ -43,6 +43,16 @@ impl Strings {
             return None;
         }
 
+        // TODO: Temporary comments for this Pull Request to draw attention to this and make a decision how to proceed.
+        //
+        // As of 3.5.0 DMTF has decided to make UTF-8 the standard for how to interpret strings.
+        // The below needs to change to interpret the byte array as UTF-8.
+        //
+        // 6.1.3 "Strings must be encoded as UTF-8 with no byte order mark (BOM). For compatibility with older SMBIOS
+        // parsers, US-ASCII characters should be used."
+        //
+        // What is confusing about the statement is that in UTF-8 the first 128 characters *are* US-ASCII.
+
         // Create an ISO-8859-1 String.  Each `u8 as char` operation maps a u8
```

Comment:
I'm thinking we might want to use https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy .

Reply:
Then again, https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8 returns Result<String, FromUtf8Error> and the error contains the full vec<u8> for inspection.  We would then refactor every struct that returns an Option<String> to now return Result<String, FromUtf8Error> .  Features such as JSON generation which depend on these strings could follow the design pattern and have a "lossy" version.  This would ensure no data is lost and the consumer of the library decides how they want to handle the situation.

Reply:
Both options make no_std support more of a pain, though I'm not sure what the best options are for utf8 in no_std environments. 
Of these two, I think I would prefer #from_utf8.

Reply:
Thanks for the no_std insights.  I haven't studied no_std design concepts and weighed the design options (abstraction layers for example).  I want the library to be of utility to as many platforms as possible and be easy to maintain / clean code.  I will need to spend time after this PR studying the problem space.

We will go with `from_utf8` (next couple days of free time) and then re-review the final design.